plugins/bulk_upload/csv/batch/BulkUploadCategoryEntryEngineCsv.php

```diff
@@ -0,0 +1,188 @@
+					break;
+			}
+
+			if(KBatchBase::$kClient->getMultiRequestQueueSize() >= $this->multiRequestSize)
+			{
+				// handle all categoryEntry objects as the partner
+				$requestResults = KBatchBase::$kClient->doMultiRequest();
+				KBatchBase::unimpersonate();
+				$this->updateObjectsResults($requestResults, $bulkUploadResultChunk);
+				$this->checkAborted();
+				KBatchBase::impersonate($this->currentPartnerId);;
+				KBatchBase::$kClient->startMultiRequest();
+				$bulkUploadResultChunk = array();
+			}
+		}
+
+		// make all the category entry actions as the partner
+		$requestResults = KBatchBase::$kClient->doMultiRequest();
+
+		KBatchBase::unimpersonate();
+
+		if(count($requestResults))
+			$this->updateObjectsResults($requestResults, $bulkUploadResultChunk);
+
+		KalturaLog::info("job[{$this->job->id}] finish updating category entries");
+	}
+
+	protected function getColumns()
+	{
+		return array(
+			self::ACTION,
+			self::ENTRY_ID,
+			self::CATEGORY_ID
+		);
+	}
+
+	protected function getUploadResultInstance()
+	{
+		return new KalturaBulkUploadResultCategoryEntry();
+	}
+
+	public function getObjectTypeTitle()
+	{
+		return self::OBJECT_TYPE_TITLE;
+	}
+
+	protected function createUploadResult($values, $columns)
+	{
+		$bulkUploadResult = parent::createUploadResult($values, $columns);
+		if (!$bulkUploadResult)
+			return;
+
+		$bulkUploadResult->bulkUploadResultObjectType = KalturaBulkUploadObjectType::CATEGORY_ENTRY;
+
+		array_walk($values, array('BulkUploadCategoryEntryEngineCsv', 'trimArray'));
+		$this->setResultValues($columns, $values, $bulkUploadResult);
+
+		$bulkUploadResult->status = KalturaBulkUploadResultStatus::IN_PROGRESS;
+
+		if (!$bulkUploadResult->action)
+		{
+			$bulkUploadResult->action = KalturaBulkUploadAction::ACTIVATE;
+		}
+
+		$bulkUploadResult = $this->validateBulkUploadResult($bulkUploadResult);
+		if($bulkUploadResult)
+		{
+			$this->bulkUploadResults[] = $bulkUploadResult;
+		}
+	}
+
+	protected function validateBulkUploadResult (KalturaBulkUploadResult $bulkUploadResult)
+	{
+		/* @var $bulkUploadResult KalturaBulkUploadResultUser */
+		if (!$bulkUploadResult->entryId || !$bulkUploadResult->categoryId)
+		{
+			$this->handleResultError($bulkUploadResult, KalturaBatchJobErrorTypes::APP, self::MISSING_COLUMN);
+
+			if($this->maxRecords && $this->lineNumber > $this->maxRecords) // check max records
+			{
+				$this->handleResultError($bulkUploadResult, KalturaBatchJobErrorTypes::APP, self::EXCEEDED_MAX_RESULTS);
+			}
+
+			if($bulkUploadResult->status == KalturaBulkUploadResultStatus::ERROR)
+			{
+				$this->addBulkUploadResult($bulkUploadResult);
+				return null;
+			}
+		}
+		return $bulkUploadResult;
+	}
+
+	protected function updateObjectsResults(array $requestResults, array $bulkUploadResults)
+	{
+		KalturaLog::info("Updating " . count($requestResults) . " results");
+		$dummy = array();

```

Comment:
This var is actually the multi request results of the updateObjectsResults. 
Please rename to a more meaningful name.